src/a11y-dialog.ts

```diff
@@ -74,6 +77,16 @@ export default class A11yDialog {
     // If the event was prevented, do not continue with the normal behavior
     if (showEvent.defaultPrevented) return this
 
+    // When opening the dialog, create a new `CloseWatcher` instance, and listen
+    // for a close event to call our `.hide(..)` method and nuking the close
+    // watcher once it’s been consumed
+    if (typeof CloseWatcher !== 'undefined') {
+      closeWatcher = new CloseWatcher()
+      closeWatcher.onclose = event => {
+        this.hide(event)
+      }
```

Comment:
I don’t know whether we should remove the watcher at that point, like this:
```suggestion
      closeWatcher.onclose = event => {
        this.hide(event)
        closeWatcher = null
      }
```

Reply:
Maybe it should be defined within local scope to ensure that it automatically gets garbage collected?